src/styl/_components/_block.styl

```diff
@@ -135,23 +136,34 @@ $block-colors ?= merge($colors-themes, {
     /* Applies theme color on title background and footer icons */
     for key, value in $block-colors
         $name = replace('theme-', '', key)
+        $theme-color = value
+        $theme-color-dark = lookup('$color-' + key + '-dark') || value
+        $theme-shadow-color = _contrast_choose($color-white, $theme-color darken($theme-color, 20%) darken($theme-color, 40%) darken($theme-color, 60%) darken($theme-color, 80%))
+        $theme-shadow-color-dark = _contrast_choose($color-white, $theme-color-dark darken($theme-color-dark, 20%) darken($theme-color-dark, 40%) darken($theme-color-dark, 60%) darken($theme-color-dark, 80%))
 
         &-{$name}
             .block-title
-                background-color value
+                background-color s('var(--color-%s)', unquote(key))
+                color $color-white
 
-                if !_contrast_check($color-white, value, 1.5)
-                    color $color-base
-                else
-                    color $color-white
-                    text-shadow 0 0 3px _contrast_choose($color-white, value darken(value, 20%) darken(value, 40%) darken(value, 60%) darken(value, 80%))
+                &
+                [data-theme=light] &
```

Comment:
Isn't this means there should be `[data-theme=light]` ? Is it possible to also specify for `:root` ?

Reply:
`.block-default .block-title`
and
`:root .block-default .block-title`
are same selector (but different spécificity)

so `&` line 149 is default mode